### PR TITLE
Feature/questions

### PR DESCRIPTION
**Pull request by guillemurano**

### CODE REVIEW

File: src/DonateTo.WebApplication/ClientApp/i18n/en.json [hunk 1]
```diff
@@ -188,7 +188,8 @@
       "Title": {
         "Donation": "Donations",
         "User": "Users",
-        "Organization": "Organization"
+        "Organization": "Organizations",
```
Comment (sculebras) [inline src/DonateTo.WebApplication/ClientApp/i18n/en.json:191]: Respect pluralization in translations.
I see that Organization translation changed to "Organizations" (perhaps now it sounds easier to only change it here).
This limits that in future we could have the translation for both Organization and Organizations without confussion.
So it should be changed the property name to Organizations and also all places where the property name is used.

File: src/DonateTo.WebApplication/ClientApp/i18n/en.json [hunk 2]
```diff
@@ -188,7 +188,8 @@
       "Title": {
         "Donation": "Donations",
```
Comment (sculebras) [inline src/DonateTo.WebApplication/ClientApp/i18n/en.json:189]: Respect pluralization in translations.
Change Donation property name to Donations and also all places where the property name is used.

File: src/DonateTo.WebApplication/ClientApp/i18n/en.json [hunk 3]
```diff
@@ -188,7 +188,8 @@
...
         "User": "Users",
-        "Organization": "Organization"
+        "Organization": "Organizations",
+        "Question": "Questions"
```
Comment (sculebras) [inline src/DonateTo.WebApplication/ClientApp/i18n/en.json:192]: Respect pluralization in translations.
Change Question property name to Questions and also all places where the property name is used.

File: src/DonateTo.WebApplication/ClientApp/i18n/en.json [hunk 4]
```diff
@@ -264,6 +265,27 @@
     "Organization": {
       "Title": "Organization"
     },
+    "Question": {
```
Comment (sculebras) [inline src/DonateTo.WebApplication/ClientApp/i18n/en.json:268]: Respect pluralization in translations.
Search for in file: "Category": "Categories" in this file
Change Category property name to Categories and also all places where the property name is used.

File: src/DonateTo.WebApplication/ClientApp/i18n/es.json [hunk 5]
```diff
@@ -188,7 +188,8 @@
       "Title": {
         "Donation": "Donaciones",
```
Comment (sculebras) [inline src/DonateTo.WebApplication/ClientApp/i18n/es.json:189]: Respect pluralization in translations.

File: src/DonateTo.WebApplication/ClientApp/i18n/es.json [hunk 6]
```diff
@@ -188,7 +188,8 @@
...
         "User": "Usuarios",
-        "Organization": "Organización"
+        "Organization": "Organizaciones",
```
Comment (sculebras) [inline src/DonateTo.WebApplication/ClientApp/i18n/es.json:191]: Respect pluralization in translations.
Change Organization property name to Organizations

File: src/DonateTo.WebApplication/ClientApp/i18n/es.json (hunk 5 above)
Comment (sculebras) [reply src/DonateTo.WebApplication/ClientApp/i18n/es.json:189]: Change Donation property name to Donations

File: src/DonateTo.WebApplication/ClientApp/i18n/es.json [hunk 7]
```diff
@@ -188,7 +188,8 @@
...
+        "Question": "Preguntas"
```
Comment (sculebras) [inline src/DonateTo.WebApplication/ClientApp/i18n/es.json:192]: Respect pluralization in translations.
Change Question property name to Questions

File: src/DonateTo.WebApplication/ClientApp/i18n/es.json [hunk 8]
```diff
@@ -264,6 +265,27 @@
     "Organization": {
       "Title": "Organización"
     },
+    "Question": {
```
Comment (sculebras) [inline src/DonateTo.WebApplication/ClientApp/i18n/es.json:268]: Respect pluralization in translations.
Search for in file: "Category": "Categories" in this file
Change Category property name to Categories

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/donations/donations-priority/donations-priority.component.ts [hunk 9]
```diff
@@ -33,8 +41,9 @@ export class DonationPriorityComponent implements OnInit {
             createdDate: new Date(),
             updateBy: '',
             updateDate: new Date(),
-            key: '1',
```
Comment (sculebras) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/donations/donations-priority/donations-priority.component.ts:36]: Consider Data population from an external source (database table, Json) and  not hardcode data for controls.
Ask @guillemurano to confirm is necessary as future improvement

Comment (guillemurano) [reply src/DonateTo.WebApplication/ClientApp/src/app/admin/donations/donations-priority/donations-priority.component.ts:36]: This is a mock inserted only for testing purposes till complete CRUD functionality is added

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/async-services/http/question.service.ts [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+import { QuestionFilter } from '../../models/filters/question-filter';
+import { PageModel } from '../../models/page.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class QuestionService extends BaseHttpClientService<QuestionModel> {
+  constructor(httpClient: HttpClient, configService: ConfigService, h1: HttpClient) {
+    const baseUrl = configService.get('baseUrl');
+    super(httpClient, baseUrl, 'api/v1/question');
```
Comment (sculebras) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/async-services/http/question.service.ts:16]: I sugget using a global var (Ex name: apiVersion) for storing version string: "api/v1/" as baseUrl.
And replace every ocurrence of using the string: "api/v1/" (Ex: other services) for  apiVersion var.

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/store/question/actions.ts [hunk 11]
```diff
@@ -0,0 +1,13 @@
+import { createAction, props } from '@ngrx/store';
+import { PageModel, QuestionModel } from '../../models';
+import { QuestionFilter } from '../../models/filters/question-filter';
+
+export const loadQuestionsPagedFiltered = createAction(
+  '[Question] Load items from server paged and filtered',
```
Comment (sculebras) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/store/question/actions.ts:6]: Is this message shown to user?
Should we use translations instead?
Same concept for the same texts in file.

**Pull request merged**